airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfig.java

```diff
@@ -126,15 +133,24 @@ public S3FormatConfig getFormatConfig() {
     return formatConfig;
   }
 
-  public AmazonS3 getS3Client() {
+  public synchronized AmazonS3 getS3Client() {
+    if (s3Client.get() == null) {
+      s3Client.set(createS3Client());
+    }
+    return s3Client.get();
+  }
+
+  public AmazonS3 createS3Client() {
+    LOGGER.info("Creating S3 client...");
+
     final AWSCredentials awsCreds = new BasicAWSCredentials(accessKeyId, secretAccessKey);
 
     if (accessKeyId.isEmpty() && !secretAccessKey.isEmpty()
         || !accessKeyId.isEmpty() && secretAccessKey.isEmpty()) {
       throw new RuntimeException("Either both accessKeyId and secretAccessKey should be provided, or neither");
     }
 
-    if (accessKeyId.isEmpty() && secretAccessKey.isEmpty()) {
+    if (accessKeyId.isEmpty()) {
```

Comment:
is this change because we already made sure that both were empty previously on line 150?